docs/root/overview.md

```diff
@@ -1,60 +1,70 @@
 # Nighthawk: architecture and key concepts
 
-## High level interaction model
-
-**Process** creates one or more **Workers**. **Worker** will run **Sequencer**,
-which in turn queries **RateLimiter** for [request-release
-timings](terminology.md#request-release-timings). When it is time to release a
-request, **BenchmarkClient** will be requested to do so by **Sequencer**.
-**BenchMarkClient** will then ask its underlying **Pool** to create a
-**StreamDecoder** for releasing the actual request. **StreamDecoder** will query the
-request data it needs to send from the configured **RequestSource**, and send it
-off. **StreamDecoder** will emit events as it progresses (pool ready,
-completion, etc), and timings will subsequently be recorded into **Statistic**
-as well as get bubbled up to **Sequencer** for tracking in-flight work and
-**Statistic** bookkeeping.
-
-**Sequencer** will query the configured **TerminationPredicates** to terminate
-when and how to terminate execution. When all **Workers** have finished,
-**Process** will collect the results from all **Workers** via
-**OutputCollector**, and run **OutputFormatter** to transform to the requested
-output format.
-
-However, it is said that pictures say more then words, so here is a block
-diagram.
-
-![Draft diagram](draft-high-level-diagram.png)
-
-[TODO(#249):add a real diagram](https://github.com/envoyproxy/nighthawk/issues/249)
-
-## Notable upcoming changes
-
-Calling out two new concepts that may get proposed in the future, and cause some
-churn in the code base as we inject them.
-
-### Phases
-
-One notable addition / change that may get proposed in the near future is the
-introduction of **Phase**. **Phase** would represent a distinct stage of an
-execution, for example a warm-up. It would then be useful to have
-per-phase reporting of latencies as well as counters and latencies.
-
-Concretely, a warm-up phase could be represented by a distinct duration
-termination predicate and a ramping rate limiter. Upon completion, the `hot`
-BenchmarkClient with its associated pool would then be live-transferred to the
-next configured phase, after which execution can continue.
-
-One other reason to have this is that it would enable remote- and/or cli-
-controlled ramping of certain test parameters by associating those to different
-phases. Termination predicates can be leveraged to immediately terminate the
-current phase after injecting a new one, allowing for real-time steering via
-gRPC and/or CLI.
-
-### Streaming parameterization and output stats
-
-Once we have phases, the gRPC service, and perhaps the CLI, would be natural
-candidates to follow up with to allow dynamic phase injection, as well as send
-back reports per phase.
+## Architecture
+
+The following diagram outlines the overall architecture of Nighthawk.
+
+![Nighthawk architecture](images/nighthawk_architecture.png)
+
```

Comment:
diagram comments:
1. The output formatter should, I believe, be returning results to the CLI.
2. Feels like we are missing some arrows on the return path - Should we indicate that the client workers are sending the responses to process?  Do they actually put them directly into the output collector?
3. (optional) - Especially if we are going to include return process arrows, can we possibly number or label the arrows?

Reply:
1. added an arrow going back to the user. While technically that goes back to the CLI (or the gRPC), adding it there seems to complicate the diagram. The return arrow crosses the API boundary, so this should be clear enough.
2. added a return arrow. Technically the workers write the stats directly, but this diagram needs a bit of abstraction to remain readable.
3. labeled most of the arrows.

Done in e4079385796871b10bbb9e7b6a0c5dee433af0e8.

Reply:
I leave it up to you how to handle this, but I will note the output formatter doesn't do anything if we're using the grpc api.  The output is always a proto there.  That's one of the reasons I clarified that they go to the CLI.  However, if you think this is a minor difference or clear enough, I defer to you.